src/storage/exec/StorageIterator.h

```diff
@@ -114,6 +116,16 @@ class SingleEdgeIterator : public StorageIterator {
             return false;
         }
 
+        // check dst
+        if (dstId_.hasValue()) {
+            auto rawDstId = NebulaKeyUtils::getDstId(planContext_->vIdLen_, iter_->key());
+            // TODO optimize the compare
+            dstId_.value().resize(rawDstId.size());
+            if (dstId_.value() != rawDstId) {
```

Comment:
Why not just push down a filter?

Reply:
So what's the filter expression?

Reply:
`edge._dst == xxx`?

Reply:
But the `xxx` is different for each vertex.It's not constant.

Reply:
I know exactly what you mean here. We need to design related query grammar. What if we need to specify rank as well, add one extra template? It seems unacceptable to me.

Reply:
If you don't want the template parameter, I think could let the StoragePlan::go accept the Row type as the input.